.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "googletest/googletest"]
```

Comment:
I think we should use `FetchContent` rather than a submodule for this. See https://github.com/microsoft/OpenCLOn12/pull/27/commits/f1ff3019e2df0911cbd7781543267ae0ba1ed88c for an example of how.

Reply:
I have checked documentation and comments and I agree with you. `FetchContent` is a better option than submodule as it's easier to initialize for the user and better track the updates. I've replaced the submodule with `FetchContent` commands.

I also noticed that your repo has a different fix for the code generation (`MTd` vs. `MDd`) issue using a one line `set(gtest_force_shared_crt...)` command and found it useful. I removed the text substitution workaround and applied your fix. Thanks!

For the `FetchContent` command, I set the tag to `master` to follow Google Test's [Live at Head](https://abseil.io/about/philosophy#upgrade-support) code management philosophy. Is that acceptable?

Reply:
Yeah, sounds good to me, thanks!